datasets/hkcancor/hkcancor.py

```diff
@@ -0,0 +1,161 @@
+# coding=utf-8
+# Copyright 2020 The HuggingFace Datasets Authors and the current dataset script contributor.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""Hong Kong Cantonese Corpus (HKCanCor)."""
+
+from __future__ import absolute_import, division, print_function
+
+import os
+import xml.etree.ElementTree as ET
+
+import datasets
+
+
+_CITATION = """\
+@article{luke2015hong,
+  author={Luke, Kang-Kwong and Wong, May LY},
+  title={The Hong Kong Cantonese corpus: design and uses},
+  journal={Journal of Chinese Linguistics},
+  year={2015},
+  pages={309-330},
+  month={12}
+}
+"""
+
+_DESCRIPTION = """\
+The Hong Kong Cantonese Corpus (HKCanCor) comprise transcribed conversations
+recorded between March 1997 and August 1998. It contains recordings of
+spontaneous speech (51 texts) and radio programmes (42 texts),
+which involve 2 to 4 speakers, with 1 text of monologue.
+
+In total, the corpus contains around 230,000 Chinese words.
+The text is word-segmented, annotated with part-of-speech (POS) tags and
+romanised Cantonese pronunciation.
+
+Romanisation scheme - Linguistic Society of Hong Kong (LSHK)
+POS scheme - Peita-Fujitsu-Renmin Ribao (PRF) corpus (Duan et al., 2000)
+"""
+
+_HOMEPAGE = "http://compling.hss.ntu.edu.sg/hkcancor/"
+
+_LICENSE = "CC BY 4.0"
+
+_URL = "http://compling.hss.ntu.edu.sg/hkcancor/data/hkcancor-utf8.zip"
+
+
+class HkCanCor(datasets.GeneratorBasedBuilder):
```

Comment:
```suggestion
class Hkcancor(datasets.GeneratorBasedBuilder):
```
this must be the camel case version of the dataset script name `hkcancor`